File: leetcode/editor/en/[2330]Valid Palindrome IV.java
2330
        Valid Palindrome IV
        2023-01-01 22:52:25

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean makePalindrome(String s) {
        int count = 0;
        for (int left = 0, right = s.length() - 1; left < right; left++, right--) {
            if (s.charAt(left) != s.charAt(right)) {
                count++;
            }
        }
        return count < 3;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2330]Valid Palindrome IV.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {boolean}
 */
var makePalindrome = function (s) {
    let count = 0;
    let left = 0;
    let right = s.length - 1;

    while (left < right) {
        if (s[left] !== s[right]) {
            count++;

            if (count > 2) {
                return false;
            }
        }

        left++;
        right--;
    }

    return true;
};
//leetcode submit region end(Prohibit modification and deletion)
